platform/SharedPtr.h

```diff
@@ -0,0 +1,215 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2006-2018 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __SHAREDPTR_H__
+#define __SHAREDPTR_H__
+
+#include <stdlib.h>
+
+#include <stdint.h>
+#include <stddef.h>
+
+/** Shared pointer class.
+  *
+  * Similar to std::shared_ptr in C++11.
+  *
+  * Usage: SharedPtr<class> POINTER(new class())
```

Comment:
Please mention that it is not a full feature shared_ptr: 
- no weak ptr
- no make_shared
- no custom deleter 
- no cast operator
- no enable_shared_from_this

Reply:
Good point!